### ISSUE TITLE
awsutils_default_vpc_deletion: DHCP options sets should also be deleted

### ISSUE THREAD

**Issue by onitake** [cloudposse/terraform-provider-awsutils#11]
## Describe the Feature

The awsutils_default_vpc_deletion deletes most resources associated with each default VPC, but it misses a resource that is also created by default in a new AWS account: DHCP option sets.

These option sets are usually missed because they have a _reverse_ dependency on a VPC, i.e. can only be deleted after it is deassociated from the VPC, or after the VPC is gone. But they should be deleted nonetheless, if a clean account without default network resources is desired.

## Use Case

I'm trying to create clean AWS accounts without any leftover resources that AWS creates by default.

## Describe Ideal Solution

1. Call https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcs.html to determine the ID of the default DHCP option set of the VPC (filter by vpc-id, return the value of result.Vpcs[0].DhcpOptionsId)
2. Call https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AssociateDhcpOptions.html with DhcpOptionsId="default" on the VPC
3. Delete the VPC
4. Call https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteDhcpOptions.html on the option set

Repeat for all regions.

## Alternatives Considered

N/A

## Additional Context

N/A

**Comment by oneingan** [comment]
This issue looks stalled and is the only piece required for have a clean pristine AWS account. is there any blocker? If it is only time maybe i could try a PR.

**Comment by ramongalate** [comment]
I agree that the dhcp could be deleted, but the dependency graph for it is not vpc -> dhcp, its dhcp -> vcp. its used by the vpc, so we can't delete it before the vpc. 

what would be nice is having a delete_default_vpc_dhcp_options

**Comment by ramongalate** [comment]
also, the vpc module that most people use for it https://github.com/terraform-aws-modules/terraform-aws-vpc, uses the default dhcp for its vpc. so deleting it with the vpc might be bad.
